File: Data Structures & Algorithms/is-palindrome/submission-1.py
class Solution:
    def isPalindrome(self, s: str) -> bool:

        beg = 0
        end = len(s) - 1

        while(beg < end):
            if s[beg].isalnum() and s[end].isalnum():
                if s[beg].lower() != s[end].lower():
                    return False
                else:
                    beg += 1
                    end -= 1
                    continue
            else:
                if not s[beg].isalnum():
                    beg += 1
                if not s[end].isalnum():
                    end -= 1
        
        return True

File: Data Structures & Algorithms/binary-search/submission-1.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int ans = binarySearch(0, nums.size()-1, target, nums);

        return ans;
    }

    int binarySearch(int beg, int end, int target, vector<int>& nums){
        if(beg > end){
            return -1;
        }

        int mid = (beg + end) /2;

        if(nums[mid] == target){
            return mid;
        }
        else if(nums[mid] < target){
            return binarySearch(mid + 1,  end,  target,  nums);
        }
        else if(nums[mid] > target){
            return binarySearch( beg,  mid - 1,  target,  nums);
        }
        return -1;
    }
};
